### PR TITLE
[front] - chore: use sparkle `Popover`

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR refactors several components in the front-end codebase to use the new `Popover` component from `sparkle` instead of the misused `DropdownMenu`. The changes are part of an ongoing effort to improve and standardize UI components across the application.

Key changes include:
- Replacing `DropdownMenu` with `Popover` in `AssistantPicker`, `AgentMessage`, `ConversationTitle`, `MessageActions`, `ActionsScreen`, `InstructionScreen`, and `Sharing` components.
- Updating the implementation to use the new `Popover` API, which provides better flexibility and accessibility.
- Adjusting the component structure to accommodate the new `Popover` usage, including changes to triggers and content rendering.

**References:**
- https://github.com/dust-tt/dust/pull/7944

## Risk

Low as they are mainly UI related

## Deploy Plan

- Deploy `sparkle` https://github.com/dust-tt/dust/pull/7944
- Deploy this PR

### CODE REVIEW

File: front/components/assistant/conversation/messages/MessageActions.tsx [hunk 1]
```diff
@@ -222,27 +217,21 @@ function EmojiSelector({
...
-        </div>
-      </DropdownMenu.Button>
-      <DropdownMenu.Items
-        width={350}
-        origin="bottomRight"
-        overflow="visible"
-        variant="no-padding"
-      >
+    <Popover
+      fullWidth={true}
```
Comment (Fraggle) [inline front/components/assistant/conversation/messages/MessageActions.tsx:221]: nit: from @flvndvd standards, shouldn't it be `fullWidth` and that's it ?

File: front/components/assistant/conversation/messages/MessageActions.tsx [hunk 2]
```diff
@@ -261,7 +250,7 @@ function EmojiSelector({
             buttonRef.current?.click();
```
Comment (Fraggle) [inline front/components/assistant/conversation/messages/MessageActions.tsx:250]: no more buttonRef  in the button ?

File: front/components/assistant/conversation/messages/MessageActions.tsx (hunk 1 above)
Comment (Fraggle) [reply front/components/assistant/conversation/messages/MessageActions.tsx:221]: are you sure it's fullwidth ? (the old one didn't have that)

File: front/components/assistant_builder/InstructionScreen.tsx [hunk 3]
```diff
@@ -462,8 +463,8 @@ function AdvancedSettings({
             </DropdownMenu>
```
Comment (Fraggle) [inline front/components/assistant_builder/InstructionScreen.tsx:463]: still there ?

File: front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx [hunk 4]
```diff
@@ -79,17 +79,16 @@ const AssistantBuilderEmojiPicker = React.forwardRef<
...
-              <Button
-                variant="tertiary"
-                icon={EmotionLaughIcon}
-                label="Pick an Emoji"
-              />
-            </div>
-          </DropdownMenu.Button>
-          <DropdownMenu.Items width={350} origin="topLeft" variant="no-padding">
+        <Popover
+          fullWidth={true}
```
Comment (Fraggle) [inline front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx:83]: same question/remark as above.

File: front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx [hunk 5]
```diff
@@ -103,36 +102,29 @@ const AssistantBuilderEmojiPicker = React.forwardRef<
...
-              />
-            </div>
-          </DropdownMenu.Button>
-          <DropdownMenu.Items width={240} origin="topLeft" variant="no-padding">
+          }
+        />
+        <Popover
+          fullWidth={true}
+          trigger={
+            <Button variant="tertiary" icon={PaintIcon} label="Pick a color" />
```
Comment (Fraggle) [inline front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx:110]: no more ref ?

**Review by Fraggle — APPROVED** [review]
LGTM overall but maybe there are a few bugs remainings.

File: front/components/assistant/conversation/messages/MessageActions.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/assistant/conversation/messages/MessageActions.tsx:221]: The `Popover` component is a new component, we didn't have one and were misusing the `DropdownMenu`. Basically passing `fullWidth` allow the `Popover` to grow, see [here](https://github.com/dust-tt/dust/blob/f9ed56e2b6d482097c8dfa914193ae21380e00e3/sparkle/src/components/Popover.tsx#L42)

File: front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx (hunk 4 above)
Comment (JulesBelveze) [reply front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx:83]: See https://github.com/dust-tt/dust/pull/7960#discussion_r1793228665

File: front/components/assistant_builder/InstructionScreen.tsx (hunk 3 above)
Comment (JulesBelveze) [reply front/components/assistant_builder/InstructionScreen.tsx:463]: Yes, it used to be a `DropdownMenu` opening a `DropdownMenu` 😅 
Now it's a `Popover` containing a `DropdownMenu`

File: front/components/assistant/conversation/messages/MessageActions.tsx (hunk 2 above)
Comment (JulesBelveze) [reply front/components/assistant/conversation/messages/MessageActions.tsx:250]: Overlooked, thanks 🙏🏼

File: front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx (hunk 5 above)
Comment (JulesBelveze) [reply front/components/assistant_builder/avatar_picker/AssistantBuilderEmojiPicker.tsx:110]: Overlooked, thx 🙏🏼

**Pull request merged**